### PR TITLE
New symsynd with inline symbolication

### PR DESCRIPTION
**Pull request by mitsuhiko**
This pull request updates the symbolication support for latest symsynd
and supports inline frames and instruction address heuristics for most frames.

This might change grouping of existing frames so we should be careful
with pushing out the update and test it thorougly first.

NOTE: will require a bump to symsynd in the getsentry repo

#nochanges

### CODE REVIEW

File: src/sentry/lang/native/symbolizer.py [hunk 1]
```diff
@@ -223,24 +266,28 @@ def symbolize_app_frame(self, frame, img):
...
+                symbolize_inlined=symbolize_inlined)
         except SymbolicationError as e:
             raise SymbolicationFailed(
                 type=EventError.NATIVE_BAD_DSYM,
                 message=six.text_type(e),
                 image=img
             )
 
-        if new_frame is None:
+        if not rv:
```
Comment (MaxBittker) [inline src/sentry/lang/native/symbolizer.py]: is rv another falsey value ever?

Comment (mitsuhiko) [inline src/sentry/lang/native/symbolizer.py]: In case inline symbolication is enabled it is an empty list here.

**Comment by mitsuhiko** [comment]
I think conceptionally we should aim for getting rid of `instruction_offset` because we are not actually in the situation to generate the correct value anyways.  I want to make a separate PR where we remove it from the UI and see if people complain.

File: src/sentry/lang/native/plugin.py [hunk 2]
```diff
@@ -421,11 +451,14 @@ def process_frame(self, frame):
             else:
                 logger.debug('Failed to symbolicate with native backend',
                              exc_info=True)
-        else:
+            return None, [raw_frame], errors
+
+        for sfrm in symbolicated_frames:
             symbol = sfrm.get('symbol_name') or \
-                new_frame.get('function') or '<unknown>'
+                frame.get('function') or '<unknown>'
```
Comment (HazAT) [inline src/sentry/lang/native/plugin.py]: One thing I noticed on the PR before ... maybe we should put the `<unkown>' string in a global constant

**Review by HazAT — CHANGES_REQUESTED** [review]

File: src/sentry/lang/native/plugin.py (hunk 2 above)
Comment (mitsuhiko) [inline src/sentry/lang/native/plugin.py]: Done.

**Review by HazAT — APPROVED** [review]

**Pull request merged**